src/App.tsx

```diff
@@ -1,26 +1,105 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from 'react';
 
-const USER_ID = 0;
+import { createTodo, deleteTodo, getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { getFilteredTodos } from './utils/getFilterdTodos';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Error } from './components/Error';
+import { TypeError } from './types/TypeError';
+import { SelectedType } from './types/SelectedType';
+import { Header } from './components/Header';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<TypeError>(
+    TypeError.DEFAULT,
+  );
+  const [selectedOption, setSelectedOption] = useState<SelectedType>(
+    SelectedType.ALL,
+  );
+  const [newTitle, setNewTitle] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [temptTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletingTodoIds, setDeletingTodoIds] = useState<number[]>([]);
+
+  const filteredTodos = getFilteredTodos(todos, selectedOption);
+
+  const handleAddTodo = (newTodo: Omit<Todo, 'id'>) => {
+    createTodo(newTodo)
+      .then(addedTodo => {
+        setTodos(currentTodos => [...currentTodos, addedTodo]);
+        setNewTitle('');
+      })
+      .catch(() => {
+        setErrorMessage(TypeError.ADDING);
+      })
+      .finally(() => {
+        setLoading(false);
+        setTempTodo(null);
+      });
+  };
+
+  const handleDeleteTodo = (todoId: number) => {
+    setLoading(true);
+    setDeletingTodoIds(currentIds => [...currentIds, todoId]);
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+      })
+      .catch(() => {
+        setErrorMessage(TypeError.DELETING);
+      })
+      .finally(() => {
+        setLoading(false);
+        setDeletingTodoIds([]);
+      });
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage(TypeError.LOADING));
```

Comment:
It's better to move the logic inside the useEffect hook to a separate function, as it improves code readability, reusability, and makes testing easier. It also helps keep useEffect focused on side effects.